src/lib.rs

```diff
@@ -1211,6 +1211,60 @@ impl<T: TensorType> Tensor<T> {
         Ok(self)
     }
 
+    /// Set one single value on the tensor.
+    ///
+    /// ```
+    /// # use tensorflow::Tensor;
+    /// let mut a = Tensor::<i32>::new(&[3, 3, 3]);
+    ///
+    /// a.set(&[0], 10);
+    /// assert_eq!(a.get(&[0]), 10);
+    ///
+    /// a.set(&[2, 2], 9);
+    /// assert_eq!(a.get(&[2, 2]), 9);
+    /// ```
+    pub fn set(&mut self, indices: &[u64], value: T) {
+        let index = self.get_index(indices);
+        self[index] = value;
+    }
+
+    /// Get one single value from the Tensor.
+    ///
+    /// ```
+    /// # use tensorflow::Tensor;
+    /// let mut a = Tensor::<i32>::new(&[3, 3, 3]);
+    ///
+    /// a.set(&[2], 5);
+    /// assert_eq!(a.get(&[2]), 5);
+    /// ```
+    pub fn get(&self, indices: &[u64]) -> T {
+        let index = self.get_index(indices);
+        self[index].clone()
+    }
+
+    /// Get the array index from rows / columns indices.
+    ///
+    /// ```
+    /// # use tensorflow::Tensor;
+    /// let a = Tensor::<f32>::new(&[3, 3, 3]);
+    ///
+    /// assert_eq!(a.get_index(&[2, 2, 2]), 26);
+    /// assert_eq!(a.get_index(&[1, 2, 2]), 25);
+    /// assert_eq!(a.get_index(&[1, 2, 0]), 7);
+    /// assert_eq!(a.get_index(&[1, 0]), 1);
+    /// ```
+    pub fn get_index(&self, indices: &[u64]) -> usize {
+        assert!(self.dims.len() >= indices.len());
```

Comment:
I'm not sure there's any real benefit to allowing users to specify only part of the index, and it's likely to cause confusion.  I would just require `self.dims.len() == indices.len()`.

Reply:
Yep I agree, done